oarepo_oidc_einfra/services/components/aai_communities.py

```diff
@@ -5,50 +5,101 @@
 # modify it under the terms of the MIT License; see LICENSE file for more
 # details.
 #
-"""AAI (perun) communities mapping"""
+"""AAI (perun) communities mapping."""
 
 import re
 
-from flask import current_app
+from invenio_access.permissions import Identity
+from invenio_communities.communities.records.api import Community
 from invenio_records_resources.services.records.components.base import ServiceComponent
-from invenio_records_resources.services.uow import Operation
+from invenio_records_resources.services.uow import Operation, UnitOfWork
+
+from oarepo_oidc_einfra.proxies import current_einfra_oidc
 
 
 class PropagateToAAIOp(Operation):
     """Operation to propagate community to AAI in a background process."""
 
-    def __init__(self, community):
+    def __init__(self, community: Community):
+        """Create a new operation."""
         self.community = community
 
-    def on_post_commit(self, uow):
+    def on_post_commit(self, uow: UnitOfWork) -> None:
+        """Propagate the community to AAI.
+
+        :param uow: unit of work
+        """
         from oarepo_oidc_einfra.tasks import synchronize_community_to_perun
 
-        synchronize_community_to_perun.delay(self.community.id)
+        synchronize_community_to_perun.delay(str(self.community.id))
 
 
 class CommunityAAIComponent(ServiceComponent):
     """Community AAI component that propagates the community to Perun."""
 
-    def create(self, identity, record=None, data=None, **kwargs):
-        """Create handler."""
+    def create(
+        self,
+        identity: Identity,
+        *,
+        record: Community,
+        data: dict,
+        **kwargs: dict,
+    ) -> None:
+        """Create handler.
+
+        This handler schedules the community to be propagated to AAI if the configuration
+        (EINFRA_COMMUNITY_SYNCHRONIZATION) allows it.
```

Comment:
Is there any purpose to this component when sync disabled? (see comment in ext)

Reply:
must be done this way because initialization of flask extensions is not ordered and thus the sync disabled setting can be introduced _after_ this component is initialized.